src/cargo/core/resolver/conflict_cache.rs

```diff
@@ -1,40 +1,98 @@
-use std::collections::{HashMap, HashSet};
+use std::collections::{BTreeMap, HashMap, HashSet};
 
 use super::types::ConflictReason;
 use core::resolver::Context;
 use core::{Dependency, PackageId};
 
+enum ConflictStore {
+    Con(BTreeMap<PackageId, ConflictReason>),
```

Comment:
The naming here was sort of confusing for me when I was initially reading this. Could a brief comment also be added about the general structure here and how `Con` is a leaf and `Map` is a parent node?